fuse/data/tokenizers/modular_tokenizer/op.py

```diff
@@ -506,3 +509,83 @@ def __call__(
                 )
 
         return sample_dict
+
+    @classmethod
+    def from_pretrained(
+        cls,
+        identifier: str,
+        pad_token: str = "<PAD>",
+        max_size: Optional[int] = None,
+        force_download: bool = False,
+        resume_download: Optional[bool] = None,
+        proxies: Optional[Dict] = None,
+        token: Optional[Union[str, bool]] = None,
+        cache_dir: Optional[Union[str, Path]] = None,
+        local_files_only: bool = False,
+        revision: Optional[str] = None,
+    ) -> "ModularTokenizerOp":
+        if not os.path.isdir(identifier):
+            # Try to download from hub
+            try:
+                # Download 'tokenizer' folder from repo
+                identifier = snapshot_download(
+                    repo_id=str(identifier),
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    proxies=proxies,
+                    resume_download=resume_download,
+                    token=token,
+                    local_files_only=local_files_only,
+                    allow_patterns="tokenizer/",
+                )
+                # Redirecting identifier to the downloaded folder
+                identifier = os.path.join(identifier, "tokenizer")
+            except Exception as e:
+                raise Exception(
+                    f"Couldn't find the checkpoint path nor download from HF hub! {identifier}"
+                ) from e
+
+        tokenizer_op = cls(
+            tokenizer_path=identifier, pad_token=pad_token, max_size=max_size
+        )
+        return tokenizer_op
+
+    def save_pretrained(self, save_directory: Union[str, Path]) -> None:
+        print(f"Saving @ {save_directory=}")
+        self._tokenizer.save(path=str(save_directory))
```

Comment:
As far as I understand, all of the information is being stored in `tokenizer`. That's why it's enough to save it.

Reply:
Correct